integration-tests/src/test/java/org/zowe/apiml/functional/gateway/ServiceHaModeTest.java

```diff
@@ -11,89 +11,211 @@
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 import org.zowe.apiml.util.TestWithStartedInstances;
 import org.zowe.apiml.util.categories.GatewayTest;
 import org.zowe.apiml.util.categories.TestsNotMeantForZowe;
 import org.zowe.apiml.util.config.RandomPorts;
 import org.zowe.apiml.util.service.VirtualService;
 
+import java.io.IOException;
 import java.util.List;
 import java.util.StringTokenizer;
 import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
 
-import static io.restassured.RestAssured.given;
+import static io.restassured.RestAssured.when;
 import static org.awaitility.Awaitility.await;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.greaterThan;
-import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.*;
 import static org.junit.jupiter.api.Assertions.fail;
 import static org.zowe.apiml.util.SecurityUtils.getConfiguredSslConfig;
 
 /**
  * Objective is to test Gateway can retry on service that is down.
- *
+ * <p>
  * 2 services are registered and after that, one is killed. Service is called through Gateway
  * and responses are inspected. Implementation returns a debug header that describes the retries.
  * The test repeats calls until it sees that request has been retried from mentioned header.
  */
 @TestsNotMeantForZowe
 @GatewayTest
-@Disabled("Flaky on different environments")
 class ServiceHaModeTest implements TestWithStartedInstances {
+
     private static final int TIMEOUT = 30;
 
-    @BeforeEach
-    void setUp() {
+    @BeforeAll
+    static void setUp() {
         RestAssured.useRelaxedHTTPSValidation();
         RestAssured.config = RestAssured.config().sslConfig(getConfiguredSslConfig());
     }
 
-    @Test
-    void givenTwoServices_whenOneServiceGoesDown_verifyThatGatewayRetriesToTheLiveOne() throws Exception {
+    private static Stream<Arguments> retryableHttpMethods() {
+        return Stream.of(
+                Arguments.of(Method.GET)
+        );
+    }
 
-        List<Integer> ports = RandomPorts.generateUniquePorts(2);
+    private static Stream<Arguments> nonRetryableHttpMethods() {
+        return Stream.of(
+                Arguments.of(Method.POST),
+                Arguments.of(Method.PUT),
+                Arguments.of(Method.DELETE),
+                Arguments.of(Method.OPTIONS)
+        );
+    }
 
-        try (
-            VirtualService service1 = new VirtualService("testHaModeService", ports.get(0));
-            VirtualService service2 = new VirtualService("testHaModeService", ports.get(1));
-            ) {
+    private static Stream<Arguments> httpMethods() {
+        return Stream.concat(retryableHttpMethods(), nonRetryableHttpMethods());
+    }
```

Comment:
Maybe add a `@SuppressWarnings("unused") // paramterized test` and add it closer to the calling method

Reply:
I don't see any warnings from IntelliJ or Sonar but I can use this annotation.

Is it possible to put it closer? Static class can't be inside inner classes @Nested. But there may be some way how to do it ...

Reply:
It could be something about intellij, this may be something about eclipse / vscode only
```
The method httpMethods() from the type ServiceHaModeTest is never used locally Java(603979894)
Stream<Arguments> org.zowe.apiml.functional.gateway.ServiceHaModeTest.httpMethods()
```